bootstrapvz/plugins/puppet/tasks.py

```diff
@@ -1,63 +1,124 @@
+from bootstrapvz.common.tools import sed_i, log_check_call
+
+# This is an updated puppet module version, introducing the puppet 4 agent
+# and also enabling you to install puppet modules on the image ready.
+# For now it ASSUMES the following:
+# - Debian Wheezy Jessie install
+# - You want the puppet 4 agent (1.9.+) installed from apt.puppetlabs.com
+#
+# Modules installed with this plugin are "--force install"
 
 
 class CheckAssetsPath(Task):
-    description = 'Checking whether the assets path exist'
+    description = 'PUPPET: Checking whether the assets path exist'
     phase = phases.validation
 
     @classmethod
     def run(cls, info):
         from bootstrapvz.common.exceptions import TaskError
         assets = info.manifest.plugins['puppet']['assets']
         if not os.path.exists(assets):
-            msg = 'The assets directory {assets} does not exist.'.format(assets=assets)
+            msg = 'PUPPET: The assets directory {assets} does not exist.'.format(assets=assets)
             raise TaskError(msg)
         if not os.path.isdir(assets):
-            msg = 'The assets path {assets} does not point to a directory.'.format(assets=assets)
+            msg = 'PUPPET: The assets path {assets} does not point to a directory.'.format(assets=assets)
             raise TaskError(msg)
 
 
 class CheckManifestPath(Task):
-    description = 'Checking whether the manifest path exist'
+    description = 'PUPPET: Checking whether the manifest path exist'
     phase = phases.validation
 
     @classmethod
     def run(cls, info):
-        from bootstrapvz.common.exceptions import TaskError
         manifest = info.manifest.plugins['puppet']['manifest']
         if not os.path.exists(manifest):
-            msg = 'The manifest file {manifest} does not exist.'.format(manifest=manifest)
+            msg = 'PUPPET: The manifest file {manifest} does not exist.'.format(manifest=manifest)
             raise TaskError(msg)
         if not os.path.isfile(manifest):
-            msg = 'The manifest path {manifest} does not point to a file.'.format(manifest=manifest)
+            msg = 'PUPPET: The manifest path {manifest} does not point to a file.'.format(manifest=manifest)
             raise TaskError(msg)
 
 
-class AddPackages(Task):
-    description = 'Add puppet package'
+class AddPuppetlabsPC1SourcesList(Task):
+    description = 'PUPPET: Adding Puppetlabs APT repo to the list of sources.'
     phase = phases.preparation
 
     @classmethod
     def run(cls, info):
-        info.packages.add('puppet')
+        if info.manifest.release == jessie:
+            info.source_lists.add('puppetlabs', 'deb http://apt.puppetlabs.com jessie PC1')
+        elif info.manifest.release == wheezy:
+            info.source_lists.add('puppetlabs', 'deb http://apt.puppetlabs.com wheezy PC1')
+        else:
+            msg = 'PUPPET: The chosen Debian release \"{system.release}\"Is not (yet) supported by Puppetlabs.com'
```

Comment:
You should add the release check in either the `validate_manifest()` function or have a separate task for this in the validation phase.